core/src/solution_submission/retry.rs

```diff
@@ -99,9 +106,44 @@ fn gas_price(estimated_price: U256, price_increase_count: u32, cap: U256) -> U25
     cap.min(U256::from(new_price as u128))
 }
 
+/// The output type of futures used in `select_all`.
+enum SelectFutureOutput {
+    SolutionSubmission(Result<(), MethodError>),
+    Timeout,
+}
+
+// Return the first future result, skipping nonce errors unless the nonce error is the last result.
+//
+// `futures` contains all solution submission attempts and optionally a timeout. Unfinished futures
+//  stay in the vector when the function returns. Must not be empty.
```

Comment:
nit: Add a `# Panics` section in the doc comments to indicate that this method panics if the futures vector is empty.